src/libstore/content-address.cc

```diff
@@ -120,8 +120,10 @@ ContentAddress parseContentAddress(std::string_view rawCa) {
 
 ContentAddressMethod parseContentAddressMethod(std::string_view caMethod)
 {
-    std::string_view asPrefix {std::string{caMethod} + ":"};
-    return parseContentAddressMethodPrefix(asPrefix);
+    std::string asPrefix = std::string{caMethod} + ":";
+    // parseContentAddressMethodPrefix takes its argument by reference
+    std::string_view asPrefixView = asPrefix;
+    return parseContentAddressMethodPrefix(asPrefixView);
```

Comment:
Does this present any advantage over

```suggestion
    return parseContentAddressMethodPrefix(asPrefix);
```

?

Reply:
It just wouldn't work, because parseContentAddressMethodPrefix takes its argument as (std::string_view &) and & won't bind to a temporary

Reply:
`std::string_view &` should be changed to `std::string_view` since there is no point in passing a `std::string_view` by reference (it's just a 2-word tuple internally). And then what @regnat suggests should work.

Reply:
Ideally, yes, would be nice to take it by value. But it seems that other code is doing something modifying the argument, I'm not very confident about what's going on there. I think, it leaves the non-consumed part in there.